api/server/sdk/alerts.go

```diff
@@ -158,3 +158,54 @@ func (g *AlertsServer) Enumerate(ctx context.Context,
 			"deadline is reached, server side func exiting")
 	}
 }
+
+// Delete implements api.OpenStorageAlertsServer for AlertsServer.
+// Input context should ideally have a deadline, in which case, a
+// graceful exit is ensured within that deadline.
+func (g *AlertsServer) Delete(ctx context.Context,
+	request *api.SdkAlertsDeleteRequest) (*api.SdkAlertsDeleteResponse, error) {
+
+	// if input has deadline, ensure graceful exit within that deadline.
+	deadline, ok := ctx.Deadline()
+	var cancel context.CancelFunc
+	if ok {
+		// create a new context that will get done on deadline
+		ctx, cancel = context.WithTimeout(ctx, deadline.Sub(time.Now()))
+		defer cancel()
+	}
+
+	group, _ := errgroup.WithContext(ctx)
+	errChan := make(chan error)
+
+	resp := new(api.SdkAlertsDeleteResponse)
+
+	queries := request.GetQueries()
+	if queries == nil {
+		return nil, status.Error(codes.InvalidArgument, "must provide a query")
+	}
+
+	filters := getFilters(queries)
+
+	// spawn err-group process.
+	group.Go(func() error {
+		return g.FilterDeleter.Delete(filters...)
+	})
+
+	// wait for err-group processes to be done
+	go func() {
+		errChan <- group.Wait()
+	}()
+
+	// wait only as long as context deadline allows
+	select {
+	case err := <-errChan:
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
```

Comment:
Use `status.Errorf(codes.Internal, "Some explanation here: %v", err)`

Reply:
Remember that the reader of these errors most likely has no idea how OpenStorage or any of its drivers works, so the explanation should be clear.